// @ionic-native/clipboard/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var Clipboard = /** @class */ (function (_super) {
    __extends(Clipboard, _super);
    function Clipboard() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Clipboard.prototype.copy = function (text) { return cordova(this, "copy", {}, arguments); };
    Clipboard.prototype.paste = function () { return cordova(this, "paste", {}, arguments); };
    Clipboard.prototype.clear = function () { return cordova(this, "clear", {}, arguments); };
    Clipboard.pluginName = "Clipboard";
    Clipboard.plugin = "cordova-clipboard";
    Clipboard.pluginRef = "cordova.plugins.clipboard";
    Clipboard.repo = "https://github.com/ihadeed/cordova-clipboard";
    Clipboard.platforms = ["Android", "iOS", "Windows Phone 8"];
Clipboard.ɵfac = function Clipboard_Factory(t) { return ɵClipboard_BaseFactory(t || Clipboard); };
Clipboard.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Clipboard, factory: function (t) { return Clipboard.ɵfac(t); } });
var ɵClipboard_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(Clipboard);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Clipboard, [{
        type: Injectable
    }], null, null); })();
    return Clipboard;
}(IonicNativePlugin));
export { Clipboard };
