// @ionic-native/file-opener/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var FileOpener = /** @class */ (function (_super) {
    __extends(FileOpener, _super);
    function FileOpener() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FileOpener.prototype.open = function (filePath, fileMIMEType) { return cordova(this, "open", { "callbackStyle": "object", "successName": "success", "errorName": "error" }, arguments); };
    FileOpener.prototype.uninstall = function (packageId) { return cordova(this, "uninstall", { "callbackStyle": "object", "successName": "success", "errorName": "error" }, arguments); };
    FileOpener.prototype.appIsInstalled = function (packageId) { return cordova(this, "appIsInstalled", { "callbackStyle": "object", "successName": "success", "errorName": "error" }, arguments); };
    FileOpener.prototype.showOpenWithDialog = function (filePath, fileMIMEType) { return cordova(this, "showOpenWithDialog", { "callbackStyle": "object", "successName": "success", "errorName": "error" }, arguments); };
    FileOpener.pluginName = "FileOpener";
    FileOpener.plugin = "cordova-plugin-file-opener2";
    FileOpener.pluginRef = "cordova.plugins.fileOpener2";
    FileOpener.repo = "https://github.com/pwlin/cordova-plugin-file-opener2";
    FileOpener.platforms = ["Android", "iOS", "Windows", "Windows Phone 8"];
FileOpener.ɵfac = function FileOpener_Factory(t) { return ɵFileOpener_BaseFactory(t || FileOpener); };
FileOpener.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: FileOpener, factory: function (t) { return FileOpener.ɵfac(t); } });
var ɵFileOpener_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(FileOpener);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FileOpener, [{
        type: Injectable
    }], null, null); })();
    return FileOpener;
}(IonicNativePlugin));
export { FileOpener };
